__tests__/integration/utils/benchmark/wallet-benchmark.util.js

```diff
@@ -0,0 +1,187 @@
+    return instances[walletId];
+  }
+
+  const walletObj = {
+    walletId,
+  };
+  if (isMultisig) walletObj.isMultisig = true;
+  instances[walletId] = walletObj;
+  return walletObj;
+}
+
+/**
+ * Wallet initialization events
+ * @readonly
+ * @enum {string}
+ */
+export const WALLET_EVENTS = {
+  startRequest: 'startRequest',
+  startResponse: 'startResponse',
+  confirmedReady: 'confirmedReady',
+  failure: 'failure',
+};
+
+export class WalletBenchmarkUtil {
+  /**
+   * Inform a wallet event
+   * @param {string} walletId
+   * @param {string} event Type of event, from WALLET_EVENTS list
+   * @param [options]
+   * @param {boolean} [options.multisig] Inform if this is a multisig wallet
+   */
+  static informWalletEvent(walletId, event, options = {}) {
+    const walletObj = getOrInitInstance(walletId, options.multisig);
+
+    switch (event) {
+      case WALLET_EVENTS.startRequest:
+        walletObj.startRequestTime = Date.now().valueOf();
+        break;
+      case WALLET_EVENTS.startResponse:
+        walletObj.startResponseTime = Date.now().valueOf();
+        walletObj.startDuration = walletObj.startResponseTime - walletObj.startRequestTime;
+        break;
+      case WALLET_EVENTS.confirmedReady:
+        walletObj.walletReadyTime = Date.now().valueOf();
+        walletObj.walletReadyDuration = walletObj.walletReadyTime - walletObj.startRequestTime;
+        break;
+      case WALLET_EVENTS.failure:
+        walletObj.failureTime = Date.now().valueOf();
+        walletObj.failureDuration = walletObj.failureTime - walletObj.startRequestTime;
+        break;
+      default:
+        console.warn(`Unknown wallet event: ${event}`);
+    }
+  }
+
+  /**
+   * Calculates a summary of the time taken to start all the wallets that are currently on the
+   * local storage.
+   * @param {string[]} [walletIds] Optional list restricting which wallet ids will be in the
+   *   summary
+   * @returns {{
+   *   [avgStartResponseTimeMultisig]: number,
+   *   [avgReadyTimeMultisig]: number,
+   *   [avgStartResponseTime]: number,
+   *   [avgReadyTime]: number,
+   *   startedWallets: number,
+   *   wallets: [{
+   *     startRequestTime: number,
+   *     startResponseTime: number,
+   *     walletReadyTime: number,
+   *     failureTime: number
+   *   }]
+   * }}
+   */
+  static calculateSummary(walletIds) {
+    const summary = {};
+    let sumResponseTime = 0;
+    let sumReadyTime = 0;
+    let sumResponseTimeMultisig = 0;
+    let sumReadyTimeMultisig = 0;
+
+    const allInstanceIds = Object.keys(instances);
+    const filteredInstanceIds = walletIds
+      ? allInstanceIds.filter(i => walletIds.includes(i)) // Filter only by the informed parameter
+      : allInstanceIds; // Retrieve all wallets
+    let amountOfWallets = 0;
+    let amountOfWalletsMultisig = 0;
+
+    for (const walletId of filteredInstanceIds) {
+      const wallet = instances[walletId];
+      if (wallet.hasFailed) continue; // Do not count failed wallets
```

Comment:
When do you set this `hasFailed` key?

Reply:
It was missing. This flag is added whenever a failure event is informed.
Fixed on [9f52708](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/9f52708e12cd76db23fcf1d693c3dd11871285b5).